state/indexer/block/kv/kv.go

```diff
@@ -24,25 +26,23 @@ var _ indexer.BlockIndexer = (*BlockerIndexer)(nil)
 // events with an underlying KV store. Block events are indexed by their height,
 // such that matching search criteria returns the respective block height(s).
 type BlockerIndexer struct {
-	store store.KVStore
+	store ds.Datastore
+
+	ctx context.Context
 }
 
-func New(store store.KVStore) *BlockerIndexer {
+func New(ctx context.Context, store ds.Datastore) *BlockerIndexer {
 	return &BlockerIndexer{
 		store: store,
+		ctx:   ctx,
 	}
 }
 
 // Has returns true if the given height has been indexed. An error is returned
 // upon database query failure.
 func (idx *BlockerIndexer) Has(height int64) (bool, error) {
```

Comment:
in some places rollmint treats height as `int64`, my best guess is that it is due to tendermint which also treats their header height as `int64`. not sure why? @tzdybal

Reply:
You're both right - there is a complete mess, caused by "compatibility" issues, with Tendermint. And it will be more messy after adapting go-header interface. We'll need a separate issue to find the best solution (I think we'll use int64 everywhere).

In this particular case (indexer) - this code is "borrowed" from Tendermint. We try to minimize the set of changes in this code.